### PR TITLE
Tosh/#37/db schema

### LINKED ISSUE #37: User model DB schema definitions
Design user entity for database. 
Ensure it contains:

- [ ] id (primary key)
- [ ] phone (unique)
- [ ] email (nullable)
- [ ] date of birth
- [ ] culture toggle
- [ ] timestamps

Perform any relevant DB migrations for the same.

### PR DESCRIPTION
**Pull request by Muriithi-Gitonga**
*This PR contains:*
- user model
- schema
- db migrations

### CODE REVIEW

File: api/db/schema.rb
Comment (otsembo) [inline api/db/schema.rb:1]: Update schema after updating migration file.

File: api/spec/models/user_spec.rb
Comment (otsembo) [inline api/spec/models/user_spec.rb:1]: Add tests for the following:

- [ ] Existence of `has_secure_password` helper method.
- [ ] All validation methods.

File: api/db/migrate/20230504174232_create_users.rb
Comment (otsembo) [inline api/db/migrate/20230504174232_create_users.rb:1]: Add the following fields:

- [ ] `password_digest` - Password hash.
- [ ]  `status` - An `enum` to indicate the account status (**Active / Inactive**).
- [ ]  Change the `culture` column and rename to `isMuslim`.
- [ ]  `date_of_birth` column should be made `non-nullable`

File: api/app/models/user.rb
Comment (otsembo) [inline api/app/models/user.rb:1]: Include the following:

- [ ] Add the `has_secure_password`  helper from BCrypt.
- [ ] Add validation helpers for `email`, `password` (_**At least 6 characters, including one uppercase and one number**_), `phone` (_**At least 9 digits, max 12 digits**_), `date_of_birth` (_**User needs to be at least 18 years of age at registration**_)

**Review by otsembo — CHANGES_REQUESTED** [review]
Hello @Muriithi-Gitonga Great Job on the PR.
I have added comments on what needs to be adjusted.
One of your checks is not passing, kindly look into that.

Ensure you pull the latest changes, I have had to update the CI script.

**Comment by Muriithi-Gitonga** [comment]
Working on the changes

**Pull request closed without merging**